tools/coredump/ebpfcode.go

```diff
@@ -12,8 +12,16 @@ package main
 // Also the tail call helper "bpf_tail_call" is overridden here, as it works in
 // tandem with the main entry point's setjmp.
 
+// todo: enable -Wunused-parameter after 1.25 upgrade https://go-review.googlesource.com/c/go/+/642196
+
 /*
-#define TESTING
+#cgo CFLAGS: -Wall -Wextra -Werror
+#cgo CFLAGS: -Wno-address-of-packed-member
+#cgo CFLAGS: -Wno-unused-label
+#cgo CFLAGS: -Wno-sign-compare
+#cgo CFLAGS: -Wno-unknown-pragmas
```

Comment:
I am not sure about this one. This one is needed because of unroll pragma. Maybe it's better to create a macro for unroll with an empty stub for core dumps. This would potentially allow detecting any other pragma warnings if any (unlikely but still).

Reply:
Not sure I understand, why should clang not recognize any pragma? If so, we can also drop that pragma, as it will be ignored anyway. And we only support compiling with clang currently.

Reply:
I would not care about pragmas here. The ebpf code is fairly small, and the main use case is the actual ebpf code. The coredump cgo build is only portions of the code. We really use pragmas for very few things only. Mainly the `unroll` for the ebpf, and those do not really matter on the coredump build.

Reply:
I think we should just not unroll when compiling with cgo. And remove (not add) the  `-Wno-unknown-pragmas`

Reply:
That's what I meant with my first comment here. And we are only focused on clang atm, just ignore gcc.

Reply:
Applying `-Wno-pass-failed` then.

Reply:
turns out we can't just ignore gcc as it is used in crosscompilation